orttraining/orttraining/python/training/onnxblock/_graph_utils.py

```diff
@@ -203,3 +201,40 @@ def get_model_parameters(model, args_not_requiring_gradient):
                 trainable_params.append(initializer)
 
     return trainable_params, non_trainable_params
+
+
+def build_graph_outputs(model, output_names):
+    """Append graph outputs for the model for the given output names.
+
+    The graph outputs are extracted from the graph value_infos and
```

Comment:
graph.value_info contains value info for intermediate outputs ... basically after shape inference value info for all the intermediate outputs  is placed in value_info and all the graph outputs is placed in output field.